File: sorting_algoritham/bubble_sort.py
def bubble_sort(a,n):
    for i in range(n):
        for j in range(0,n-i-1):
            if (a[j] > a[j+1]):
                a[j],a[j+1]=a[j+1],a[j]

    return


array=[]
n=int(input("enter the size of array:"))
for i in range(n):
    ele=int(input())
    array.append(ele)

print("before sorting the array is:")
for i in range(len(array)):
    print(array[i],end=" ")

bubble_sort(array,n)

print("\nafter sorting the array is:")
for i in range(len(array)):
    print(array[i],end=' ' )

File: sll/bubble_sort.c
#include<stdio.h>
#define SIZE 50
void bubble_sort(int [],int);
int main()
{
	int n,array[SIZE],i;
	printf("enter the array size:");
	scanf("%d",&n);
	printf("enter the array elements:\n");
	for(i=0;i<n;i++)
	{
		scanf("%d",&array[i]);
	}
	printf("before sort the array elements are:\n");
	for(i=0;i<n;i++)
	{
		printf("%d\t",array[i]);
	}
	void(*fptr)(int [],int)=bubble_sort;
	fptr(array,n);

}
void bubble_sort(int a[],int n)
{
	int i,j,temp;
	for(i=0;i<n;i++)
	{
		for(j=0;j<n-i-1;j++)
		{
			if(a[j] > a[j+1])
			{
				temp=a[j];
				a[j]=a[j+1];
				a[j+1]=temp;
			}
		}
	}
	printf("\nafter sort the array elements are:\n");
        for(i=0;i<n;i++)
                printf("%d\t",a[i]);
	printf("\n");
}
